contracts/MetaIdentityManager.sol

```diff
@@ -0,0 +1,217 @@
+pragma solidity 0.4.11;
+import "./Proxy.sol";
+
+
+contract MetaIdentityManager {
+  uint adminTimeLock;
+  uint userTimeLock;
+  uint adminRate;
+  address relay;
+
+  event IdentityCreated(
+    address indexed identity,
+    address indexed creator,
+    address owner,
+    address indexed recoveryKey);
+
+  event OwnerAdded(
+    address indexed identity,
+    address indexed owner,
+    address instigator);
+
+  event OwnerRemoved(
+    address indexed identity,
+    address indexed owner,
+    address instigator);
+
+  event RecoveryChanged(
+    address indexed identity,
+    address indexed recoveryKey,
+    address instigator);
+
+  event MigrationInitiated(
+    address indexed identity,
+    address indexed newIdManager,
+    address instigator);
+
+  event MigrationCanceled(
+    address indexed identity,
+    address indexed newIdManager,
+    address instigator);
+
+  event MigrationFinalized(
+    address indexed identity,
+    address indexed newIdManager,
+    address instigator);
+
+  mapping(address => mapping(address => uint)) owners;
+  mapping(address => address) recoveryKeys;
+  mapping(address => mapping(address => uint)) limiter;
+  mapping(address => uint) migrationInitiated;
+  mapping(address => address) migrationNewAddress;
+
+  modifier onlyAuthorized() {
+    if (msg.sender == relay || checkMessageData(msg.sender)) _;
+    else throw;
+  }
+
+  modifier onlyOwner(address identity, address sender) {
+    if (owners[identity][sender] > 0 && (owners[identity][sender] + userTimeLock) <= now ) _ ;
```

Comment:
Yes, I think this makes sense for clarity. I will update now. Here and in IdentityManager